sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -128,21 +128,29 @@ namespace Azure { namespace Data { namespace Tables {
      */
     Response<Models::DeleteEntityResult> DeleteEntity(
         Models::TableEntity const& tableEntity,
-        Core::Context const& context = {});
+        Core::Context const& context = {}) const;
 
     /**
-     * @brief Upsert specified entity in a table.
+     * @brief Update or insert specified entity in a table.
      *
-     * @param tableEntity The TableEntity to upsert.
-     * @param upsertKind Upsert kind(update/merge), default update.
+     * @param tableEntity The TableEntity to update or insert.
      * @param context for canceling long running operations.
      * @return Upsert entity result.
      */
-    Response<Models::UpsertEntityResult> UpsertEntity(
+    Response<Models::UpdateEntityResult> UpdateInsertEntity(
         Models::TableEntity const& tableEntity,
-        Models::UpsertKind upsertKind = Models::UpsertKind::Update,
-        Core::Context const& context = {});
+        Core::Context const& context = {}) const;
 
+    /**
+     * @brief Merge or insert the specified entity in a table.
+     *
+     * @param tableEntity The TableEntity to merge or insert.
+     * @param context for canceling long running operations.
+     * @return Upsert entity result.
```

Comment:
Doc comment doesn't match the return type. Copy/pasta typo

```suggestion
     * @return Merge entity result.
```

Reply:
good catch